src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -187,10 +187,14 @@ private CommandResult executeCommand(String commandText) throws CommandException
             }
 
             return commandResult;
-        } catch (CommandException | ParseException e) {
+        } catch (ParseException e) {
             logger.info("Invalid command: " + commandText);
             resultDisplay.setFeedbackToUser(e.getMessage());
             throw e;
+        } catch (CommandException e) {
+            logger.info("An error occured while executing this command: " + commandText);
+            resultDisplay.setFeedbackToUser(e.getMessage());
+            throw e;
         }
```

Comment:
I don't think this change is related to the issue at hand. It should be a separate PR.
Also, I think the following tweak is enough (no need for two catch blocks).
```suggestion
        } catch (CommandException | ParseException e) {
            logger.info("An error occured while executing this command: " + commandText);
            resultDisplay.setFeedbackToUser(e.getMessage());
            throw e;
        }
```

Reply:
I separated the 2 because in the log file, any command that triggered CommandException or ParseException will log the line "Invalid Command: %s", even though the command is valid.

Perhaps we can standardise to a general term "An error occured while executing this command: " instead of invalid command?

Reply:
>any command that triggered CommandException or ParseException previously will log the line "Invalid Command: %s", even though the command is valid.

I agree, that doesn't sound right.

> Perhaps we can standardise to a general term "An error occured while executing this command: " instead of invalid command as suggested?

Yes, that's what I'm suggesting as well.